src/cmd/ksh93/bltins/whence.c

```diff
@@ -283,8 +283,9 @@ static_fn int whence(Shell_t *shp, char **argv, int flags) {
                 if (flags & V_FLAG) {
                     if (*cp != '/') {
                         if (!np && (np = nv_search(name, shp->track_tree, 0))) {
+                            const char *command_path = ((struct pathcomp *)np->nvalue.cp)->name;
```

Comment:
Rather than the explicit cast just do `... = np->nvalue.pathcomp->name`.

P.S., I really, really, want to get rid of all these explicit casts. Especially in the case of `np->nvalue`. They're dangerous and hide bugs. And in the case of this struct member I want to add a corresponding enum that indicates the type of the last value assigned so that we can add asserts when using the value to verify the type that was stored is what is expected where it is used.